### PR TITLE
Performance-Analytics

### PR DESCRIPTION
**Pull request by RCAutomatizacion**
Please leave opinions about the repository update.

### CODE REVIEW

File: docs/Installation.md [hunk 1]
```diff
@@ -54,158 +72,212 @@ Delay Select Screen:
...
 
-## Configure Data Service
-Steps are created for an asset as aspects in the data service. An asset represent the stations and for each asset the aspects represents the steps. 
+## Configure IIH Essentials
+
+Steps are created for an asset as aspects in IIH Essentilas and automatically applied in Performance Insight. An asset represents, for this example, a car production "Station". and for each asset the aspects represents the steps. 
 
 ![Dataservice Structure](graphics/Dataservice_Struktur.jpg)
 
 Performance Insight use this structure of assets and aspects to visualize the data in a later step. Open the web interface of your IED and launch the Data Service app.
```
Comment (mattDoerr) [inline docs/Installation.md:173]: IIH

File: README.md [hunk 2]
```diff
@@ -17,16 +17,31 @@ Below you can find the structure of this use case:
...
+The 'Step Time Analysis' dashboard of the Edge App 'Performance Insight' is used to assess the efficiency of the implemented sequential control systems. First, it needs to be synchronized with the asset model of the plant, representing the control systems in operation. This synchronization allows for the establishment of a reference duration for each step in the process. Then, the configured (or expected) step time is compared with the actual measured time during operation. 
 
+This comparison makes it easier to identify specific steps in the production process that are experiencing delays, thereby enabling more targeted and efficient corrective actions.
+
+This specific example shows how an exemplary production line is connected, the data from the plant is transferred to the edge system and evaluated there. This is shown using the step time analysis of a fictitious automobile production with 5 assembly stations and randomly delayed steps.
```
[the remark this answers is not in the document]
Comment (RCAutomatizacion) [reply README.md]: Done

File: README.md [hunk 3]
```diff
@@ -52,35 +67,37 @@ TIA and PLC:
...
+*	SIMATIC S7 Connector V2.0.0-1
+* OPC UA Connector V2.0.1-0
+* Common Import Converter V2.0.0-1
+* Common Connector Configurator V1.9.0-11
+* Registry Service V1.9.0-0
+*	Databus V2.3.1
+*	Databus Configurator V 2.3.1-4
+*	IIH Essentials V1.9.0
+*	Performance Insight V1.16.1
+*	WEB Browser (Google Chrome)
```
[the remark this answers is not in the document]
Comment (RCAutomatizacion) [reply README.md]: Done.

File: README.md [hunk 4]
```diff
@@ -52,35 +67,37 @@ TIA and PLC:
...
 -	Configure PLC project in TIA Portal
 - Configure PLC connections in Industrial Edge
   - Configure Databus
-  - Configure S7 Connector and OPC UA Connector
-- Configure Data Service
-  - Configure the adapter
-  - Configure an asset with variables
+  - Configure PLC Connectors
+- Configure IIH Essentials
+  - Configure the Databus
```
[the remark this answers is not in the document]
Comment (RCAutomatizacion) [reply README.md]: Done

File: docs/graphics/DataFlow.png
[the remark this answers is not in the document]
Comment (RCAutomatizacion) [reply docs/graphics/DataFlow.png:1]: Done

File: docs/Installation.md [hunk 5]
```diff
@@ -5,13 +5,31 @@
...
+
+1. **OPC UA Connector**: Configure the OPC UA Connector to establish communication with PLC 1 and PLC 2. Ensure that the OPC UA servers on these PLCs are set up to allow for data exchange.
+
+2. **S7 Connector**: Set up the S7 Connector to facilitate data transfer from PLC 3 using the S7+ protocol, and from PLC 4 and PLC 5 via the standard S7 protocol.
+
+3. **Databus Configuration**: Integrate the MQTT-Broker to allow for a seamless data flow from the OPC UA and S7 Connectors to the higher-level Industrial Edge Applications.
+
+4. **IIH Essentials**: Install and configure the IIH Essentials application to ensure that the core industrial data from your PLCs can be effectively harnessed and utilized by the Edge Device. This setup is crucial for enabling comprehensive data collection and subsequent analysis within the Performance Insights application.
+
+5. **Performance Insights Application Setup**: Within the Performance Insights application, configure the 'Step Time Analysis' feature by linking it to the respective asset models, which reflect the status of the sequential control steps from the PLCs.
```
[the remark this answers is not in the document]
Comment (RCAutomatizacion) [reply docs/Installation.md]: Done

File: docs/Installation.md [hunk 6]
```diff
@@ -54,158 +72,212 @@ Delay Select Screen:
...
-Deploy the OPC UA Connector configuration.
+Deploy the OPC UA Connector by clicking "Deploy". After deployment the "Bus Adaptor" and the "Data Source" status of both PLCs should have green icon, like so:
+
+![OPC UA7](graphics/OPCUA7.png)
+
+If any issue is presented when configuring the OPC UA Connector, check [documentation](https://support.industry.siemens.com/cs/document/109811051/opc-ua-connector?dti=0&lc=en-US)
+
+### Configure S7 Connector
+
+In this section, the communication with the **third, fourth and fifth PLCs** are configured using the S7 Connector; however, the communication is set up with the S7+ protocol.
```
[the remark this answers is not in the document]
Comment (RCAutomatizacion) [reply docs/Installation.md:139]: Only S7+

File: docs/Installation.md [hunk 7]
```diff
@@ -54,158 +72,212 @@ Delay Select Screen:
...
+![S7+0](graphics/S7+0.png)
 
-![S7 PLC3](graphics/S7_Connector_PLC3.png)
+Enter your PLC3 details as shown in the image and click "Add":
 
-Also add the same tags from PLC3. Tag browsing also works with the S7+ protocol or add the tags manually.
+![S7+1](graphics/S7+1.png)
 
-PLC4 and PLC5 are used as data sources with the S7 protocol:
+Use the browsing tool to add the same tags as in previous PLCs, similar to the method used during the OPC UA Connector configuration:
```
[the remark this answers is not in the document]
Comment (RCAutomatizacion) [reply docs/Installation.md]: Done

[the remark this answers is not in the document]
Comment (RCAutomatizacion) [reply docs/Installation.md]: Done

File: docs/Installation.md [hunk 8]
```diff
@@ -54,158 +72,212 @@ Delay Select Screen:
...
+![OPC UA3](graphics/OPC_UA3.png)
 
-![S7 PLC2](graphics/S7_Connector_PLC2.png)
+Select and add the tags that are named "DB_HMI"."ARG1_Seq1_S1" to "DB_HMI"."ARG1_Seq1_S19". Also add the String variable for the product "DB_Process_Var"."Car_Type_inProduction_Text":
 
-Add the needed tags for PLC2 in the same way. Add the tags for the steps and the tag for the product.
+![OPC UA5](graphics/OPC_UA5.png)
 
-Edit the settings:
+This variables indicate if the respective step is active right now.
```
[the remark this answers is not in the document]
Comment (RCAutomatizacion) [reply docs/Installation.md]: Done

File: docs/Installation.md [hunk 9]
```diff
@@ -54,158 +72,212 @@ Delay Select Screen:
...
-Repeat this process for every station.
+### Configure Aspects
+
+Now, it's time to create the Steps that will be displayed in the **Step Time Analysis** dashboard on Performance Insights app. For every step, the PLC should supply a tag indicating the status of the step's activity.
+
+For each station, Step 2 is designated as the initial step, and its aspect type is assigned as 'StepChainAnalysis_InitialStep'. To do this, go to *Station 1 > Aspects > Add Aspect* and type the "Step 2" settings:
+
+![Station1Aspect](graphics/Station1Aspect.png)
+
+For the remaining steps (3rd to 11th), assign the aspect type 'StepChainAnalysis_InitialStep', as illustrated:
```
[the remark this answers is not in the document]
Comment (RCAutomatizacion) [reply docs/Installation.md]: Done

File: docs/Installation.md [hunk 10]
```diff
@@ -54,158 +72,212 @@ Delay Select Screen:
...
+- As the factor, you specify a percentage that is added to the initial value.
+- Click "​Apply to all steps​".
+
+​A limit is calculated for selected steps on the basis of the respective limit values and entered in the corresponding column for limit values. To save the changes and switch to the dashboard view, click "​Save​". ​If you want to continue working in the same view after saving, click "​Save​ & ​Continue​". To exit the view without saving, click "​Cancel​".
 
 ### Show step time analysis
 
-When you have created steps for the asset in the Data Service and when you have defined the limits then the step time analysis is automatically 
-displayed in the "Step time analysis" dashboard.
+When you have created steps for the asset in the Data Service and when you have defined the limits then the step time analysis is automatically displayed in the "Step time analysis" dashboard.
```
[the remark this answers is not in the document]
Comment (RCAutomatizacion) [reply docs/Installation.md]: Done

File: docs/Installation.md [hunk 11]
```diff
@@ -54,158 +72,212 @@ Delay Select Screen:
...
-* Selecting a time period for which the step time analysis is to be displayed
-* Downloading a report in CSV format
-* Switching to the limit definition in the Asset Configuration
-* Displaying the steps with highest fault rate
-* Sequence overview: Display and selection of a specific sequence for detailed display
-* "Show all sequences"
-* Sequence details: Displaying all sequences in the selected time period and defined steps. When selecting a single step, the Trend view of the step opens.
+The image above shows the Step Time Analysis overview screen where five sequences have been completed. In every sequence, only the 7th step took longer than the expected completion time, marking it as faulty in red.
+
+With the help of the "Overview" screen you can get all information about the sequences of a specific product for a specified time period. When the step is gray the step time is within the defined limits. A step displayed in red is outside the defined limits.
```
[the remark this answers is not in the document]
Comment (RCAutomatizacion) [reply docs/Installation.md:279]: You're right. However, the intention was not to explain it twice. May you please check the modification I made? Please confirm if you find the modification necessary. If not, I'll only leave this paragraph:

_With the help of the "Overview" screen you can get all information about the sequences of a specific product for a specified time period. When the step is gray the step time is within the defined limits. A step displayed in red is outside the defined limits._

File: docs/Installation.md (hunk 1 above)
Comment (RCAutomatizacion) [reply docs/Installation.md]: Done

File: README.md [hunk 12]
```diff
@@ -17,16 +17,31 @@ Below you can find the structure of this use case:
...
+The 'Step Time Analysis' dashboard of the Edge App 'Performance Insight' is used to assess the efficiency of the implemented sequential control systems. First, it needs to be synchronized with the asset model of the plant, representing the control systems in operation. This synchronization allows for the establishment of a reference duration for each step in the process. Then, the configured (or expected) step time is compared with the actual measured time during operation. 
 
+This comparison makes it easier to identify specific steps in the production process that are experiencing delays, thereby enabling more targeted and efficient corrective actions.
+
+This specific example shows how an exemplary production line is connected, the data from the plant is transferred to the edge system and evaluated there. This is shown using the step time analysis of a fictitious automobile production with 5 assembly stations and randomly delayed steps.
 
-![overview1](docs/graphics/overview1.png)
```
[the remark this answers is not in the document]
Comment (RCAutomatizacion) [reply README.md:23]: Do you mean to remove them from the graphics folder?

File: docs/graphics/DASHBOARD.png
[the remark this answers is not in the document]
Comment (RCAutomatizacion) [reply docs/graphics/DASHBOARD.png:1]: No, I just deleted it from the graphics folder.

**Review by mattDoerr — APPROVED** [review]

**Pull request merged**